interfaces/builtin/connectivity_observe.go

```diff
@@ -0,0 +1,156 @@
+`)
+
+var connectivityObserveConnectedSlotAppArmor = []byte(`
+# Description: Allow access to NetworkingStatus service.
+
+dbus (receive)
+    bus=system
+    path=/com/ubuntu/connectivity1/NetworkingStatus{,/**}
+    interface=org.freedesktop.DBus.*,
+`)
+
+var connectivityObserveConnectedPlugAppArmor = []byte(`
+# Description: Allow using NetworkingStatus service.
+
+#include <abstractions/dbus-strict>
+
+# Allow all access to NetworkingStatus service
+dbus (receive, send)
+    bus=system
+    interface=com.ubuntu.connectivity1.NetworkingStatus{,/**}
+    path=/com/ubuntu/connectivity1/NetworkingStatus,
+
+dbus (send)
+    bus=system
+    path=/com/ubuntu/connectivity1/NetworkingStatus{,/**}
+    interface=org.freedesktop.DBus.*,
+`)
+
+var connectivityObservePermanentSlotDBus = []byte(`
+<policy user="root">
+    <allow own="com.ubuntu.connectivity1.NetworkingStatus"/>
+    <allow send_destination="com.ubuntu.connectivity1.NetworkingStatus"/>
+</policy>
+
+<policy context="default">
+    <deny own="com.ubuntu.connectivity1.NetworkingStatus"/>
+
+    <deny send_destination="com.ubuntu.connectivity1.NetworkingStatus"/>
+</policy>
+
+<limit name="max_replies_per_connection">1024</limit>
+<limit name="max_match_rules_per_connection">2048</limit>
+`)
+
+type ConnectivityObserveInterface struct{}
+
+func (iface *ConnectivityObserveInterface) Name() string {
+	return "connectivity-observe"
+}
+
+func (iface *ConnectivityObserveInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+func (iface *ConnectivityObserveInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		snippet := []byte(connectivityObserveConnectedPlugAppArmor)
+		old := []byte("###PLUG_SECURITY_TAGS###")
+		var new []byte
+		new = slotAppLabelExpr(slot)
+		snippet = bytes.Replace(snippet, old, new, -1)
+		return snippet, nil
+	}
+	return nil, nil
+}
+
+func (iface *ConnectivityObserveInterface) PermanentSlotSnippet(slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityDBus:
+		return connectivityObservePermanentSlotDBus, nil
+	case interfaces.SecurityAppArmor:
+		snippet := []byte(connectivityObservePermanentSlotAppArmor)
+		old := []byte("###SLOT_SECURITY_TAGS###")
+		var new []byte
+		new = slotAppLabelExpr(slot)
+		snippet = bytes.Replace(snippet, old, new, -1)
+		return snippet, nil
+	}
+	return nil, nil
+}
+
+func (iface *ConnectivityObserveInterface) ConnectedSlotSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		snippet := []byte(connectivityObserveConnectedSlotAppArmor)
+		old := []byte("###SLOT_SECURITY_TAGS###")
```

Comment:
This should be '###PLUG_SECURITY_TAGS###' since you want to use this in connectivityObserveConnectedSlotAppArmor: